float32[3] integrated_angles